### ISSUE TITLE
How to enforce `enum` validation on a specific property?

### ISSUE THREAD

**Issue by aMacDawg** [tdegrunt/jsonschema#369]
Use-case: a property `channelType` can only expect the enums 'youtube' and 'twitch', any other values to the `channelType` key should return a validation error.

JSON Schema snippet:

```json
let schema = { //...trunc
"properties": {
        "channelType": {
            "type": "string",
            "default": "",
            "title": "The channelType Schema",
            "items": {
                "type": "string",
                "enum": [
                    "youtube",
                    "twitch"
                ]
            }
        }
//...trunc
}
```

sample code:

```js
var validator = new Validator();
//req.query is the json obj formatted by express.js from inbound http requests
var result = validator.validate(req.query, schema, {nestedErrors: true});
```

Actual outcome: the channelType value could equal anything else ('facebook', 'sesame-street') and the validator will **not** return an error as expected

According to this [stackoverflow](https://stackoverflow.com/questions/30924271/correct-way-to-define-array-of-enums-in-json-schema) post, the schema is formatted correctly for enums, and the [npm-jsonschema docs](https://www.npmjs.com/package/jsonschema) did not specifically reference handling enums

**Comment by aMacDawg** [comment]
Ok, I had to do word salad with my own question, but found another [stackoverflow post](https://stackoverflow.com/questions/26532137/jsonschema-multiple-values-for-string-property) that got closer to what I was trying to get at - validation on a `string` field as opposed to validation on an `array` field which is what the `enum` approach seems to be for.

tl;dr - this works:
```json
"channelType": {
            "type": "string",
            "default": "",
            "title": "The channelType Schema",
            "oneOf": [
                {"enum": ["youtube", "twitch"]}
            ]
        }
```

Closing this 🎉
